js/script.js

```diff
@@ -1,44 +1,93 @@
 'use strict';
 
 function titleClickHandler(event){
-    event.preventDefault();
-const clickedElement = this;
+  event.preventDefault();
+  const clickedElement = this;
   console.log('Link was clicked!');
   console.log(event);
 
   /* [DONE] remove class 'active' from all article links  */
-const activeLinks = document.querySelectorAll('.titles a.active');
+  const activeLinks = document.querySelectorAll('.titles a.active');
 
-for(let activeLink of activeLinks){
-  activeLink.classList.remove('active');
-}
+  for(let activeLink of activeLinks){
+    activeLink.classList.remove('active');
+  }
 
   /* [DONE] add class 'active' to the clicked link */
   clickedElement.classList.add('active');
   console.log('clickedElement:', clickedElement);
 
   /* [DONE] remove class 'active' from all articles */
-const activeArticles = document.querySelectorAll('.posts .active');
+  const activeArticles = document.querySelectorAll('.posts .active');
 
-for(let activeArticle of activeArticles){
-  activeArticle.classList.remove('active');
-}
+  for(let activeArticle of activeArticles){
+    activeArticle.classList.remove('active');
+  }
 
   /* [DONE] get 'href' attribute from the clicked link */
-const articleSelector = clickedElement.getAttribute('href');
+  const articleSelector = clickedElement.getAttribute('href');
   console.log(articleSelector);
 
   /* [DONE] find the correct article using the selector (value of 'href' attribute) */
-const targetArticle = document.querySelector(articleSelector);
+  const targetArticle = document.querySelector(articleSelector);
   console.log(targetArticle);
 
   /* [DONE] add class 'active' to the correct article */
   targetArticle.classList.add('active');
   
 }
 
-const links = document.querySelectorAll('.titles a');
 
-for(let link of links){
-  link.addEventListener('click', titleClickHandler);
-}
\ No newline at end of file
+const optArticleSelector = '.post',
+      optTitleSelector = '.post-title',
+      optTitleListSelector = '.titles';
+
+function generateTitleLinks(){
+
+  /* [DONE] remove contents of titleList */
+  const titleList = document.querySelector('.list.titles');
+  console.log('list found', titleList);
+  titleList.innerHTML ='';
+
+  /* [DONE] for each article */
+  const articles = document.querySelectorAll(optArticleSelector);
+  console.log('articles found', articles);
+
+  let html = '';
+
+  for(let article of articles){
+    
+    /* [DONE] get the article id */
+    const articleId = article.getAttribute('id');
+    //console.log('found id', articleId);
+
+    /* [DONE] find the title element */
+    /* [DONE] get the title from the title element */
+    const articleTitle = article.querySelector(optTitleSelector).innerHTML;
+    //console.log('found title', articleTitle);
+
+    /* [DONE] create HTML of the link */
+    const linkHTML = '<li><a href="#' + articleId + '"><span>' + articleTitle +'</span></a></li>';
```

Comment:
Chcesz spróbować zrobić to używając [string interpolation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#string_interpolation). Porównaj, który zapis jest bardziej czytelny ;)